signing/signer.go

```diff
@@ -13,6 +13,15 @@ import (
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
+type domain byte
+
+const (
+	ATX domain = iota
+	BEACON
+	BALLOT
+	HARE
```

Comment:
@poszu do we need to add domain for poet? there was something about validation

Reply:
i think poet forward the whole opaque blob for a node to verify the challenge. and it's done in `ChallengeVerifier`.
is that what you are referring to?

Reply:
i remember there was a discussion about signing something for the poet using smesher key, if so we need to use domain for that signature as well